src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -288,6 +289,11 @@ impl StreamSink {
                         })
                         .try_collect::<_, _, RwError>()?
                 }
+            } else if properties.get(CREATE_TABLE_IF_NOT_EXISTS) == Some(&"true".to_owned())
+                && sink_type == SinkType::Upsert
+                && from_properties.is_empty()
```

Comment:
The `from_properties` is actually `downstream_pk`. Shall we rename to the correct one?